src/main/java/com/ledikom/service/BotService.java

```diff
@@ -199,6 +213,28 @@ private void executeAdminActionOnMessageReceived(final RequestFromAdmin requestF
 
         if (splitStringsFromAdminMessage.get(0).equals(AdminMessageToken.NEWS.label)) {
             sendNewsToUsers(requestFromAdmin.getPhotoPath(), splitStringsFromAdminMessage);
+        } else if (splitStringsFromAdminMessage.get(0).equals(AdminMessageToken.COUPON.label)) {
+            createNewCoupon(requestFromAdmin.getPhotoPath(), splitStringsFromAdminMessage);
+        }
+    }
+
+    private void createNewCoupon(final String photoPath, final List<String> splitStringsFromAdminMessage) {
```

Comment:
rework to be exactly like in sendNewsToUser method where NewsFromAdmin object is created in the first line to incapsulate the logic of this splitString and to not use list.get(index) here directly, but use human readable field names of an object eg `CouponFromAdmin`